### PR TITLE
Add support for Kemp LoadMaster

### PR DESCRIPTION
**Pull request by tumd**
I haven't tested this extensively, but I've verified that all the relevant fields seem to export correctly, and added suitable overrides to the values that was in need of conversion.

Based on a blogpost by Bas Roovers.
https://onpremisys.nl/2020/01/06/monitoring-kemp-loadmaster-using-snmp-and-prometheus

### CODE REVIEW

File: generator/generator.yml [hunk 1]
```diff
@@ -362,6 +362,64 @@ modules:
...
+      - ssCpuIdle
+      - memory
+      - 1.3.6.1.4.1.12196.13.0 # VSdesc
+      - 1.3.6.1.4.1.12196.13.1 # VSentry
+      - 1.3.6.1.4.1.12196.13.2 # RSentry
+    lookups:
+      - source_indexes: [vSidx]
+        lookup: 1.3.6.1.4.1.12196.13.1.1.13 # vSName
+    overrides:
+      daemonState:
```
Comment (brian-brazil) [inline generator/generator.yml:386]: Does this change over time? Usually things called "state" would be statesets, not info.

There's other states too below.

**Review by brian-brazil — COMMENTED** [review]
It looks like the MIB is invalid per netsnmp.

File: generator/generator.yml (hunk 1 above)
Comment (tumd) [reply generator/generator.yml:386]: I suppose the "state"'s won't change over time in the MIB's. Not entirely sure however. It seem like some of the "state"s has changed in the past. `vSState` for example, seem to have removed one of it's elements at some point (3 is missing);
```
vSState OBJECT-TYPE
  SYNTAX    INTEGER {
    inService(1),
    outOfService(2),
    disabled(4),
    sorry(5),
    redirect(6),
    errormsg(7),
    securityDown(8)
  }
```
`rSState` is similar to `vSState` in missing the 3:rd element.
`daemonState` & `haState` are simpler in that regard.

What is your recommendation?

**Comment by tumd** [comment]
What would you suggest regarding the invalid timestamps? I guess the preferred would be to ask the vendor to correct the MIB's upstream.
Is there any other way around this?

File: generator/generator.yml (hunk 1 above)
Comment (SuperQ) [reply generator/generator.yml:386]: The "change over time" referrers to the device, not the MIB. Think about ethernet link state (changes over time when you plug/un-plug a cable) vs ethernet port type (gige, 10gige, sfp) that doesn't really change over time.

Comment (tumd) [reply generator/generator.yml:386]: Ah, I see. :)
In that case, they all may change on the device over time.

**Comment by SuperQ** [comment]
Strange, I don't get a parse error on net-snmp 5.8.

**Comment by brian-brazil** [comment]
We'd first have to check the RFCs to see what's going on here, this isn't an error I've come across before.

**Comment by tumd** [comment]
What I've found the [RFC2578](https://tools.ietf.org/html/rfc2578#section-2) (ExtUTCTime) quite clearly states the format of the value for `LAST-UPDATED` & `REVISION`, which the upstream vendor MIB's doesn't comply with. I haven't found any newer RFC's replacing this section in RFC2578.
I'll reach out to the vendor and ask them to revise.

**Comment by brian-brazil** [comment]
That looks to be it alright, they've included second which aren't permitted.

**Comment by SuperQ** [comment]
One option would be to add a small patch script to the Makefile to fix up the strings.

File: generator/Makefile [hunk 2]
```diff
@@ -155,6 +158,15 @@ $(MIBDIR)/KEEPALIVED-MIB:
 	@echo ">> Downloading KEEPALIVED-MIB"
 	@curl $(CURL_OPTS) -o $(MIBDIR)/KEEPALIVED-MIB $(KEEPALIVED_URL)
 
+$(MIBDIR)/.kemp-lm:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading Kemp LM MIBs to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(KEMP_LM_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) *.txt
+	@sed -i -E "s/([0-9]{12})[0-9]{2}Z/\1Z/" $(MIBDIR)/*.RELEASE-B100-MIB.txt
```
Comment (SuperQ) [inline generator/Makefile:166]: Would you mind adding a comment here that this is a workaround? Plus a minor suggestion for the regexp.

```suggestion
	# Workaround invalid timestamps.
	@sed -i -E 's/"([0-9]{12})[0-9]{2}Z"/"\1Z"/' $(MIBDIR)/*.RELEASE-B100-MIB.txt
```

**Comment by brian-brazil** [comment]
Was there any work back from the vendor? I'd rather not get into the game of altering vendor-provided MIBs if we don't have to.

File: generator/Makefile (hunk 2 above)
Comment (tumd) [reply generator/Makefile]: I agree on your suggestions and have updated the commit.
Initially I had a similar comment included locally, but was annoyed/confused of the fact that `make mibs` printed the comment, thus wasn't confident enough to include it.

**Comment by tumd** [comment]
At last Friday (13'th) the vendor said;
> The MIB issue has be recognized as a bug and I'm just waiting on the development team to provide a date for the fix.
In other words a fix may come tomorrow or in 6 months.

I totally understand that you don't want such workarounds in the code. Let me know and I'll revert it.

**Comment by SuperQ** [comment]
I think the sed patch is small and simple enough that we can keep it. We're not changing any functionality of the MIB, just working around a syntax issue.

**Review by SuperQ — APPROVED** [review]
LGTM

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**